src/main/java/app/commands/AddContactCommand.java

```diff
@@ -2,51 +2,29 @@
 import app.sql.dao.mysql.ContactDaoImpl;
+import app.sql.dao.mysql.PhoneDaoImpl;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.text.ParseException;
-import java.text.SimpleDateFormat;
-import java.util.Date;
 
 public class AddContactCommand implements ActionCommand {
     @Override
     public String execute(HttpServletRequest request, HttpServletResponse response) {
         try {
-            String name = request.getParameter("name");
-            String surname = request.getParameter("surname");
-            String familyName = request.getParameter("familyName");
-            String email = request.getParameter("email");
-            String sex = request.getParameter("sex");
-            String citizenship = request.getParameter("citizenship");
-            String relationship = request.getParameter("relationship");
-            String webSite = request.getParameter("webSite");
-            String currentJob = request.getParameter("currentJob");
-            String jobAddress = request.getParameter("jobAddress");
-            String residenceCountry = request.getParameter("residenceCountry");
-            String residenceCity = request.getParameter("residenceCity");
-            String residenceStreet = request.getParameter("residenceStreet");
-            String residenceHouseNumber = request.getParameter("residenceHouseNumber");
-            String residenceApartmentNumber = request.getParameter("residenceApartmentNumber");
-            String index = request.getParameter("index");
-            java.sql.Date sqlStartDate;
-            try {
-                Date date = new SimpleDateFormat("yyyy-MM-dd").parse(request.getParameter("date"));
-                sqlStartDate = new java.sql.Date(date.getTime());
-            } catch (ParseException e) {
-                sqlStartDate = null;
-            }
-
-            Contact contact = new Contact(null, email, name, surname,
-                    familyName, sqlStartDate, sex, citizenship,
-                    relationship, webSite, currentJob,
-                    jobAddress, residenceCountry, residenceCity, residenceStreet, residenceHouseNumber,
-                    residenceApartmentNumber, index);
-            ContactDaoImpl.getInstance().save(contact);
-        } catch (AppException e) {
+            // TODO: check if user tries to submit phoneNumber that already exists. Save the other ones?
+            Contact contact = ContactDaoImpl.getInstance().save(ContactService.getContactParameters(request));
+            PhoneDaoImpl.getInstance().saveAll(PhoneService.getAllPhoneNumbers(request, contact.getId()));
+        } catch (AppException | IllegalArgumentException e) {
+            request.setAttribute("error", e.getMessage());
+        } catch (Exception ex) {
+            request.setAttribute("error", "Unknown error occurred");
 
         }
         return new ShowAllContactsCommand().execute(request, response);
```

Comment:
Looks weird in a case if an error occurred. You just add error message to request and after that return `ShowAllContactsCommand().execute(request, response)`. It would be better to pass error message in your `AppException` and wrap any command invocation in servletDispatcher, extract message from error and set it to request/response.